docs/sequence.rst

```diff
@@ -0,0 +1,672 @@
+    foldr (+) 0 (Seq.fromList [0, 1, 2])
+    > 3
+
+.. TIP:: You can use the `OverloadedLists
+	 <https://ghc.haskell.org/trac/ghc/wiki/OverloadedLists>`_ extension so
+	 you don't need to write ``fromList [1, 2, 3]`` everywhere, instead you
+	 can just write ``[1, 2, 3]`` and if the function is expecting a
+	 sequence it will be converted automatically! The code here will
+	 continue to use ``fromList`` for clarity though.
+
+
+Importing Sequence
+------------------
+
+When using ``Sequence`` in a Haskell source file you should always use a
+``qualified`` import becasue it exports names that clash with the standard
+Prelude (you can import the type constructor and some operators on their own
+though!).
+
+::
+
+    import Data.Sequence (Seq(..), (<|), (|>), (><))
+    import qualified Data.Sequence as Seq
+
+
+Common API Functions
+--------------------
+
+.. NOTE::
+   ``fromList [some,sequence,elements]`` is how a ``Seq`` is printed.
+
+Construction and Conversion
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create an empty sequence
+""""""""""""""""""""""""
+
+::
+
+    Seq.empty :: Seq a
+    Seq.empty = ...
+
+:seq:`empty` creates a sequence with zero elements.
+
+::
+
+    Seq.empty
+    > fromList []
+
+
+Create a sequence with one element (singleton)
+""""""""""""""""""""""""""""""""""""""""""""""
+
+::
+
+    Seq.singleton :: a -> Seq a
+    Seq.singleton x = ...
+
+:seq:`singleton` creates a sequence with the single element ``x`` in it.
+
+::
+
+    Seq.singleton "containers"
+    > fromList ["containers"]
+
+    Seq.singleton 1
+    > fromList [1]
+
+Create a sequence with the same element repeated
+""""""""""""""""""""""""""""""""""""""""""""""""
+
+::
+
+    Seq.replicate :: Int -> a -> Seq a
+    Seq.replicate n x = ...
+
+:seq:`replicate` creates a sequence with same element ``x`` repeated ``n``
+times.
+
+::
+
+    Seq.replicate 0 "hi"
+    > fromList []
+
+    Seq.replicate 3 "hi"
+    > fromList ["hi","hi","hi"]
+
+Create a sequence from a list
+"""""""""""""""""""""""""""""
+
+::
+
+    Seq.fromList :: [a] -> Seq a
+    Seq.FromList xs = ...
+
+:seq:`fromList` creates a sequence containing the elements of the list
+``xs``. Sequences allow duplicate so all elements will be included in the order
+given.
+
+::
+
+    Seq.fromList ["base", "containers", "QuickCheck"]
+    > fromList ["base","containers","QuickCheck"]
+
+    Seq.fromList [0, 1, 1, 2, 3, 1]
+    > fromList [0,1,1,2,3,1]
+
+Adding to an existing sequence
+""""""""""""""""""""""""""""""
+
+::
+
+    (<|) :: a -> Seq a -> Seq a
+    x <| xs = ...
+
+    (|>) :: Seq a -> a -> Seq a
+    xs |> x = ...
+
+    (><) :: Seq a -> Seq a -> Seq a
+    l >< r = ...
+
+- ``x <| xs`` places the element ``x`` at the beginning of the sequence ``xs``..
+
+- ``xs |> x`` places the element ``x`` at the end of the sequence ``xs``.
+
+- ``l >< r`` combines the two sequences ``l`` and ``r`` together.
+
+
+Create a list from a sequence
+"""""""""""""""""""""""""""""
+
+::
+
```

Comment:
There is no need to write this out by hand using `foldr`. Just use `Data.Foldable.toList`!